src/autoscaler/api/publicapiserver/public_api_handler.go

```diff
@@ -197,62 +203,29 @@ func (h *PublicApiHandler) DetachScalingPolicy(w http.ResponseWriter, r *http.Re
 		writeErrorResponse(w, http.StatusBadRequest, ErrorMessageAppidIsRequired)
 		return
 	}
+
 	logger := h.logger.Session("DetachScalingPolicy", lager.Data{"appId": appId})
 	logger.Info("Deleting policy json", lager.Data{"appId": appId})
-	err := h.policydb.DeletePolicy(r.Context(), appId)
-	if err != nil {
+
+	if err := h.policydb.DeletePolicy(r.Context(), appId); err != nil {
 		logger.Error("Failed to delete policy from database", err)
 		writeErrorResponse(w, http.StatusInternalServerError, "Error deleting policy")
 		return
 	}
+
 	logger.Info("Deleting schedules")
-	err = h.schedulerUtil.DeleteSchedule(r.Context(), appId)
-	if err != nil {
+	if err := h.schedulerUtil.DeleteSchedule(r.Context(), appId); err != nil {
 		logger.Error("Failed to delete schedule", err)
 		writeErrorResponse(w, http.StatusInternalServerError, "Error deleting schedules")
 		return
 	}
 
 	if h.bindingdb != nil && !reflect.ValueOf(h.bindingdb).IsNil() {
```

Comment:
We can probably remove this condition as it was only there for use with the no longer supported "builtin" mode.

Reply:
fix in c9c5e86ac